assets/chat/js/autocomplete.js

```diff
@@ -80,6 +80,9 @@ function timeoutHelpers(ac){
 function updateHelpers(ac){
     ac.chat.ui.toggleClass('chat-autocomplete-in', ac.results.length > 0)
     ac.ui.toggleClass('active', ac.results.length > 0)
+    if (ac.results.length === 0) {
+        ac.container.css('left', 0)
+    }
```

Comment:
I spent some time reviewing the code and found the issue. We don't need to rely on this hack.

The source of the autocomplete UI positioning bug is in [this block](https://github.com/destinygg/chat-gui/blob/56722a850b49287f559a353f975d845ac4e8d7d0/assets/chat/js/autocomplete.js#L95-L99).

When the left edge of the autocomplete result two positions to the left of the selected result is off screen to the left, we adjust the `<ul>`'s offset (via the `left` property) such that the left edge of that result is touching the left edge of the chat window. If the first or second result is selected, the `<ul>` is simply positioned such that the first result touches the left edge.

The issue is [here](https://github.com/destinygg/chat-gui/blob/56722a850b49287f559a353f975d845ac4e8d7d0/assets/chat/js/autocomplete.js#L98). The `left` property of the `<li>` elements are relative to their parent `<ul>`, so they're always positive. Giving the `<ul>` a positive offset shifts it off to the right, sometimes off screen. Simply changing `ac.container.css('left', $(e).position().left)` to `ac.container.css('left', -$(e).position().left)` accounts for this and fixes the issue. We can now navigate left and right through the autocomplete results with our mouse/finger, as intended.

I'll go ahead and merge if you add this to your PR. 😊

Reply:
Oh, nice! I'm not gonna lie, I've found the positioning code a bit confusing, which is why I opted for the quick hack.

I'll remove it and add what you asked NODDERS Thank you!